sematic/ui/packages/common/src/pages/RunDetails/contextMenus/RunSectionMenu.tsx

```diff
@@ -0,0 +1,81 @@
+import { useContext, useMemo, useCallback } from "react";
+import ContextMenu from "src/component/ContextMenu";
+import { useRootRunContext } from "src/context/RootRunContext";
+import SnackBarContext from "src/context/SnackBarContext";
+import { useCancelRun, useRerun } from "src/hooks/resolutionHooks";
+
+const ANCHOR_OFFSET = {x: 13, y: -38};
+
+interface RunSectionActionMenuProps {
+    anchorEl: HTMLElement | null;
+}
+
+function RunSectionActionMenu(props: RunSectionActionMenuProps) {
+    const { anchorEl } = props;
+
+    const { rootRun, resolution } = useRootRunContext();
+
+    const { setSnackMessage } = useContext(SnackBarContext);
+
+    const [rerunState, rerun] = useRerun(rootRun!.id, rootRun!.id);
+    const [cancelRunState, cancel] = useCancelRun(rootRun!.id);
+
+    const cancelEnabled = useMemo(
+        () => {
+            if (!rootRun) {
+                return false;
+            }
+            if (cancelRunState.loading) {
+                return false;
+            }
+            return !["FAILED", "NESTED_FAILED", "RESOLVED", "CANCELED"].includes(
+                rootRun!.future_state
+            );
+        }, [rootRun, cancelRunState]
+    );
+
+    const rerunEnable = useMemo(
+        () => !!resolution?.container_image_uri && !rerunState.loading,
+        [resolution, rerunState]
+    );
+
+    const onCopyShareClick = useCallback(() => {
+        navigator.clipboard.writeText(window.location.href);
+        setSnackMessage({ message: "Resolution link copied" });
+    }, [setSnackMessage]);
+
+    const commands = useMemo(() => {
+        return [
+            {
+                title: "Rerun",
+                disabled: !rerunEnable,
+                onClick: async () => {
+                    try {
+                        rerun();
+                    } catch (error) {
+                        setSnackMessage({ message: "Failed to trigger rerun." });
```

Comment:
I know this is adapted existing code, but please also include the error message, to improve debuggability.

Reply:
Noted. However, I think the problem here is that the snack bar dialog box does not have room to display a text which is too long. That's maybe why the original author made such a compromise. 

I guess looking at the network trace is fine for now. However, if this becomes a frequent issue in the future, we can adjust the UI element used later.